### PR TITLE
Added Next.js QuickStart [SDK-2233]

### PR DESCRIPTION
**Pull request by Widcket**
This PR adds the beta QuickStart for the new Next.js SDK.

### CODE REVIEW

File: articles/_includes/_logout_url.md [hunk 1]
```diff
@@ -2,9 +2,7 @@
 
 ### Configure Logout URLs
 
-A logout URL is a URL in your application that Auth0 can return to after the user has been logged out of the authorization server. This is specified in the `returnTo` query parameter.
-
-The logout URL for your app must be added to the **Allowed Logout URLs** field in your [Application Settings](${manage_url}/#/applications). If this field is not set, users will be unable to log out from the application and will get an error.
+A logout URL is a URL in your application that Auth0 can return to after the user has been logged out of the authorization server. This is specified in the `returnTo` query parameter. The logout URL for your app must be added to the **Allowed Logout URLs** field in your [Application Settings](${manage_url}/#/applications). If this field is not set, users will be unable to log out from the application and will get an error.
```
Comment (Widcket) [inline articles/_includes/_logout_url.md:5]: This is a short paragraph that does not make sense to have split, and it adds unnecessary vertical length to the QS.

File: articles/_includes/_new_app.md [hunk 2]
```diff
@@ -1,9 +1,11 @@
 ## Configure Auth0
 ### Get Your Application Keys
 
-When you signed up for Auth0, a new application was created for you, or you could have created a new one.
+When you signed up for Auth0, a new application was created for you, or you could have created a new one. You will need some details about that application to communicate with Auth0. You can get these details from the [Application Settings](${manage_url}/#/applications) section in the Auth0 dashboard.
```
Comment (Widcket) [inline articles/_includes/_new_app.md:4]: Same as above.

File: articles/_includes/_new_app.md [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 
-You will need some details about that application to communicate with Auth0. You can get these details from the [Application Settings](${manage_url}/#/applications) section in the Auth0 dashboard.
+<% if(typeof hideDashboardScreenshot === 'undefined' || hideDashboardScreenshot !== true) { %>
+![App Dashboard](/media/articles/dashboard/client_settings.png)
+<% } %>
```
Comment (Widcket) [inline articles/_includes/_new_app.md:8]: It makes more sense to have the screenshot of the Dashboard right below where it's mentioned. It also provides a little break from the repetition of all those `You`s.

File: articles/quickstart/native/_includes/_ios_config.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ You need the following information:
 * **Domain**
 
 ::: note
-If you download the sample from the top of this page, these details are filled out for you. If you have more than one application in your account, the sample comes with the values for your **Default App**.
```
Comment (Widcket) [inline articles/quickstart/native/_includes/_ios_config.md:12]: That is no longer accurate, as you can select which app you want to use.

File: articles/quickstart/native/android/_includes/_credentials.md [hunk 5]
```diff
@@ -18,5 +18,5 @@ Edit your `res/values/strings.xml` file as follows:
 ```
 
 ::: note
-If you download the sample from the top of this page, these details are filled out for you. If you have more than one application in your account, the sample comes with the values for your **Default App**.
```
Comment (Widcket) [inline articles/quickstart/native/android/_includes/_credentials.md:21]: Same as above.

File: articles/quickstart/spa/_includes/_auth_note_dev_keys.md [hunk 6]
```diff
@@ -1,5 +1,7 @@
 <!-- markdownlint-disable MD041 -->
 
 :::note
-Auth0 enables the Google social provider by default on new tenants and offers you developer keys to test logging in with [social identity providers](/connections/identity-providers-social). However, these developer keys have some limitations that may cause your application to behave differently. For more details on what this behavior may look like and how to fix it, consult the [Test Social Connections with Auth0 Developer Keys](/connections/social/devkeys#limitations-of-developer-keys) document.
+Auth0 enables the Google social provider by default on new tenants and offers you developer keys to test logging in with [social identity providers](/connections/identity-providers-social). However, these developer keys have some limitations that may cause your application to behave differently.
+
+For more details on what this behavior may look like and how to fix it, consult the [Test Social Connections with Auth0 Developer Keys](/connections/social/devkeys#limitations-of-developer-keys) document.
```
Comment (Widcket) [inline articles/quickstart/spa/_includes/_auth_note_dev_keys.md:6]: In this case it's a long paragraph that benefits from a line break.

File: articles/quickstart/webapp/_includes/_getting_started.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <!-- markdownlint-disable MD041 -->
 
 ::: note
-**New to Auth?** Learn [How Auth0 works](/overview), how it [integrates with Regular Web Applications](/architecture-scenarios/application/web-app-sso) and which [protocol](/flows/concepts/auth-code) it uses.
+**New to Auth?** Learn [How Auth0 works](/overview), how it [integrates with Regular Web Applications](/architecture-scenarios/web-app-sso) and which [protocol](/flows/concepts/auth-code-pkce) it uses.
```
Comment (adamjmcgrath) [inline articles/quickstart/webapp/_includes/_getting_started.md]: A lot of the Web App SDKs use `code`, perhaps we should just link to `/flows/concepts`

File: articles/quickstart/webapp/nextjs/index.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  setup: server-platforms/nextjs/setup
+  use: server-platforms/nextjs/use
+sdk:
+  name: nextjs-auth0
+  url: https://github.com/auth0/nextjs-auth0/
+github:
+  org: auth0-samples
+  repo: auth0-nextjs-samples
+  branch: master
+next_steps:
```
Comment (adamjmcgrath) [inline articles/quickstart/webapp/nextjs/index.yml]: Let's remove the `next_steps` section and do what we did with Express: https://github.com/auth0/docs/blob/master/articles/quickstart/webapp/express/01-login.md#whats-next

## What's next?
We put together a few examples of how to use [nextjs-auth0](https://github.com/auth0/nextjs-auth0) in more advanced use cases:

- [Protecting a Server Side Rendered (SSR) Page](https://github.com/auth0/nextjs-auth0/blob/beta/EXAMPLES.md#protecting-a-server-side-rendered-ssr-page)
- [Protecting a Client Side Rendered (CSR) Page](https://github.com/auth0/nextjs-auth0/blob/beta/EXAMPLES.md#protecting-a-client-side-rendered-csr-page)
- [Protect an API Route](https://github.com/auth0/nextjs-auth0/blob/beta/EXAMPLES.md#protect-an-api-route)
- [Access an External API from an API Route](https://github.com/auth0/nextjs-auth0/blob/beta/EXAMPLES.md#access-an-external-api-from-an-api-route)

File: articles/quickstart/webapp/nextjs/download.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+http://localhost:3000/api/auth/callback
+```
+
+2) Set **Allowed Logout URLs** in the [Application Settings](${manage_url}/#/applications/${account.clientId}/settings) to
+```text
+http://localhost:3000
+```
+3) Make sure [Node.JS LTS](https://nodejs.org/en/download/) is installed and execute the following commands in the root directory:
+```bash
+npm run install && npm start
```
Comment (adamjmcgrath) [inline articles/quickstart/webapp/nextjs/download.md]: for a Next.js app, this is probably `npm install && npm run dev`

**Comment by adamjmcgrath** [comment]
@Widcket - have you been able to run this locally? If so, can you attach a screenshot, I haven't been able to run the docs project locally and the deploys don't seem to be working

File: articles/quickstart/webapp/_includes/_getting_started.md (hunk 7 above)
Comment (Widcket) [reply articles/quickstart/webapp/_includes/_getting_started.md]: `/flows/concepts` doesn't exist, changing to `/flows`.

File: articles/quickstart/webapp/nextjs/download.md (hunk 9 above)
Comment (Widcket) [reply articles/quickstart/webapp/nextjs/download.md]: Chwnged in https://github.com/auth0/docs/pull/9557/commits/e932a625798bedc2b280ff2606c57b780ac4ef15.

File: articles/quickstart/webapp/nextjs/index.yml (hunk 8 above)
Comment (Widcket) [reply articles/quickstart/webapp/nextjs/index.yml]: Changed in https://github.com/auth0/docs/pull/9557/commits/e932a625798bedc2b280ff2606c57b780ac4ef15.

File: articles/quickstart/webapp/_includes/_getting_started.md (hunk 7 above)
Comment (Widcket) [reply articles/quickstart/webapp/_includes/_getting_started.md]: Changed in https://github.com/auth0/docs/pull/9557/commits/e932a625798bedc2b280ff2606c57b780ac4ef15.

**Comment by Widcket** [comment]
> @Widcket - have you been able to run this locally? If so, can you attach a screenshot, I haven't been able to run the docs project locally and the deploys don't seem to be working

@adamjmcgrath here's the working preview link: https://auth0content-pr-9557.herokuapp.com/docs/quickstart/webapp/nextjs

**Comment by Widcket** [comment]
Had to restart the dynos many times, it seems like the docs app is taking way too long to start:

**Comment by Widcket** [comment]
Had to restart the dynos many times, it seems like the docs app is taking way too long to start:

Screen Shot 2021-01-08 at 11 08 03

**Comment by Widcket** [comment]
@adamjmcgrath however, the quickstart doesn't show up in the qs list: https://auth0content-pr-9557.herokuapp.com/docs/quickstart/webapp, and I'm not sure why

**Comment by adamjmcgrath** [comment]
> the quickstart doesn't show up in the qs list

@Widcket - that's expected, Beta QSs don't go in the list

**Review by adamjmcgrath — APPROVED** [review]

**Comment by adamjmcgrath** [comment]
LGTM, we'll merge it shortly after the Beta release

**Comment by davidpatrick** [comment]
@Widcket there is a manual and annoying way to update the heroku app boottime timeouts https://tools.heroku.support/limits/boot_timeout

**Comment by davidpatrick** [comment]
after you udpate the timeout, you should be able to do a restart/redeploy

**Comment by Widcket** [comment]
@davidpatrick thanks, just set it to 120 secs.

**Pull request merged**